src/pages/invoices/common/hooks/useProductColumns.ts

```diff
@@ -16,47 +16,133 @@ export function useProductColumns() {
   const company = useCurrentCompany();
   const [columns, setColumns] = useState<string[]>([]);
 
+  const getColumnIndex = (
+    columnKey: string,
+    pdfColumns: string[],
+    currentColumns: string[]
+  ) => {
+    const columnIndex = pdfColumns.findIndex(
+      (variable) => variable === columnKey
+    );
+
+    const adjustedColumnIndex =
+      columnIndex > -1 ? columnIndex : currentColumns.length;
+
+    return adjustedColumnIndex;
+  };
+
   useEffect(() => {
-    // We need to clone the product columns to local object,
-    // because by default it's frozen.
```

Comment:
Deleting comments for no reason it's not always good. I rarely write them because I feel TypeScript can express what we're trying to do pretty well, but when such complex logic and lifecycles are in place, it's never bad leaving them step by step. It's for yourself, a few years in the future. 

Comments might not be up-to-date with refactoring, but they still might be helpful.

Reply:
Yeah, I agree with you. Considering the slightly more complex logic, I have left your comments as they are and added some additional helpful comments regarding the additional logic implemented in the pull request.